### PR TITLE
Transformer optimizations

### PR DESCRIPTION
**Pull request by tbennun**
A set of optimizations for the transformer application:

- [x] Synthetic data reader
- [x] Loss function optimization
- [x] Channelwise Softmax optimization
- [x] ~~Layer Normalization parity with PyTorch~~

### CODE REVIEW

File: applications/nlp/transformer/train.py [hunk 1]
```diff
@@ -99,53 +86,25 @@ def make_model(
...
+    # using values that will never contribute to the cross-entropy loss
+    labels = lbann.Select(labels, lbann.Identity(labels), value=pad_index, if_false=(vocab_size + 1))
+
+    # Compute cross-entropy
+    loss = lbann.CrossEntropy(preds, labels, use_labels=True)
 
     # Construct model
     metrics = []
-    callbacks = [lbann.CallbackPrint(), lbann.CallbackTimer()]
+    callbacks = [lbann.CallbackPrint(), lbann.CallbackTimer(), lbann.CallbackProfiler()]
```
Comment (tbennun) [inline applications/nlp/transformer/train.py:107]: we might want to remove this change

**Review by ndryden — APPROVED** [review]
Looks good to me.

File: applications/nlp/transformer/train.py (hunk 1 above)
Comment (bvanessen) [reply applications/nlp/transformer/train.py:107]: I think that we can leave it in for now.
